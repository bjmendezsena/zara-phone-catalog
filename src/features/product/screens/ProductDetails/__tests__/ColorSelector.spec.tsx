import { render, getHtml, mockProductDetailsResponse } from "@/tests";
import {
  ColorSelector,
  ColorSelectorProps,
} from "@/features/product/screens/ProductDetails/ColorSelector";

describe(`<${ColorSelector.name}/>`, () => {
  const defaultProps: ColorSelectorProps = {
    colorOptions: mockProductDetailsResponse.colorOptions,
    selectedColor: mockProductDetailsResponse.colorOptions[0].name,
    onColorChange: jest.fn(),
  };

  it("renders without crashing", () => {
    const { container } = render(<ColorSelector {...defaultProps} />);
    expect(getHtml(container)).toMatchSnapshot();
  });

  it("displays the correct number of color options", () => {
    const { getAllByRole } = render(<ColorSelector {...defaultProps} />);
    const colorOptions = getAllByRole("button");
    expect(colorOptions).toHaveLength(defaultProps.colorOptions.length);
  });

  it("highlights the selected color option", () => {
    const { getByTestId } = render(<ColorSelector {...defaultProps} />);
    const testId = `color-option-${defaultProps.selectedColor}`;
    const selectedOption = getByTestId(testId);
    expect(selectedOption).toHaveClass("border-primary");
  });
  it("calls onColorChange when a color option is clicked", () => {
    const { getByTestId } = render(<ColorSelector {...defaultProps} />);
    const testId = `color-option-${defaultProps.colorOptions[1].name}`;
    const colorOption = getByTestId(testId);
    colorOption.click();
    expect(defaultProps.onColorChange).toHaveBeenCalledWith(
      defaultProps.colorOptions[1]
    );
  });
});
