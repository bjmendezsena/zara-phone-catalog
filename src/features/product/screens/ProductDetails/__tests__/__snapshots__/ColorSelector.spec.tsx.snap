// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<ColorSelector/> renders without crashing 1`] = `
"<div>
  <div
    class="flex flex-col gap-4"
  >
    <h3
      class="text-sm mb-4 uppercase tracking-wide"
    >
      COLOR. 
      <span
        class="uppercase"
      >
        Pick your favourite
      </span>
    </h3>
    <div
      class="flex gap-3"
    >
      <button
        aria-label="Select Obsidiana color"
        class="w-[24px] h-[24px] border-1 transition-all cursor-pointer p-[2px] border-primary"
        data-testid="color-option-Obsidiana"
      >
        <div
          aria-hidden="true"
          class="w-full h-full"
          style="background-color: rgb(0, 0, 0);"
        />
      </button>
      <button
        aria-label="Select Porcelana color"
        class="w-[24px] h-[24px] border-1 transition-all cursor-pointer p-[2px] border-gray-300 hover:border-primary"
        data-testid="color-option-Porcelana"
      >
        <div
          aria-hidden="true"
          class="w-full h-full"
          style="background-color: rgb(245, 245, 245);"
        />
      </button>
      <button
        aria-label="Select Celeste color"
        class="w-[24px] h-[24px] border-1 transition-all cursor-pointer p-[2px] border-gray-300 hover:border-primary"
        data-testid="color-option-Celeste"
      >
        <div
          aria-hidden="true"
          class="w-full h-full"
          style="background-color: rgb(135, 206, 235);"
        />
      </button>
    </div>
    <span
      class="text-xs uppercase tracking-wide min-h-[15px]"
    >
      Obsidiana
    </span>
  </div>
</div>"
`;
